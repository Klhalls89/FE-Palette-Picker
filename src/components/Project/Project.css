.project-area {
  text-align: center;
}

.proj-name {
  text-align: center;
}

.saved-palette-disp {
  display: flex;
  align-items: center;
  justify-content: center;
  border: solid #fff 3px;
}

.saved-palette-disp h3 {
  font-family: 'Courgette', cursive;
}

.saved-color1:hover,
.saved-color2:hover,
.saved-color3:hover,
.saved-color4:hover,
.saved-colors5:hover {
  transform: scale(1.2);
}

.far {
  margin: 5px;
  font-size: 38px;
  padding: 0px 10px;
  margin: auto;
}

.far:hover {
  color: grey;
  transform: scale(1.2);
}

.fas {
  margin: 5px;
  font-size: 36px;
  padding-left: 17px;
  margin: auto;
}

.fas:hover {
  color: grey;
  transform: scale(1.2);
}

.saved-colors {
  grid-area: colors;
  display: flex;
}

.saved-colors div {
  background-color: pink;
  height: 50px;
  width: 50px;
  margin: 5px;
}

.saved-palette-disp:hover {
  border: solid #3C3C3C 3px;
  cursor: pointer;
}
