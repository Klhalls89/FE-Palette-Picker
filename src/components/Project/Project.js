import React from 'react';

const Project = ({ 
  palette_title, project_title, 
  id, color_1, color_2, color_3, color_4, color_5
}) => {

  if (id) { 
    return (
      <article className="project-area">
        <h2 className="proj-name">
          {project_title}
        </h2>
        <div className="saved-palette-disp">
          <h3 className="pal-name">
            {palette_title}
          </h3>
          <i className="fas fa-trash-alt"></i>
          <section className="saved-colors">
            <div style={{backgroundColor: `#${color_1}`}} 
              className="saved-color1">
            </div>
            <div style={{backgroundColor: `#${color_2}`}} 
              className="saved-color2">
            </div>
            <div style={{backgroundColor: `#${color_3}`}} 
              className="saved-color3">
            </div>
            <div style={{backgroundColor: `#${color_4}`}} 
              className="saved-color4">
            </div>
            <div style={{backgroundColor: `#${color_5}`}} 
              className="saved-color5">
            </div>
          </section>
        </div>
      </article>
    );
  } else {
    return (
      <div>
        <h1></h1>
      </div>
    );
  }


  
};
export default Project;