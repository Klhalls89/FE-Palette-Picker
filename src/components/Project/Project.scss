  .project-area {

    .proj-name {
      text-align: center;
    } 
    
    .saved-palette-disp {
      display: flex;
      align-items: center;
      justify-content: center;
      border: solid #fff 3px;

      h3 {
        font-family: $accent-font;
      }
      .saved-color1:hover {
        transform: scale(1.2)
      }
      .saved-color2:hover {
        transform: scale(1.2)
      }
      .saved-color3:hover {
        transform: scale(1.2)
      }
      .saved-color4:hover {
        transform: scale(1.2)
      }
      .saved-color5:hover {
        transform: scale(1.2)
      }
      .far {
        margin: 5px;
        font-size: 38px;
        padding: 0px 10px;
        margin: auto;
        &:hover {
          color: grey;
          transform: scale(1.2)
        }
      }
      .fas {
        margin: 5px;
        font-size: 36px;
        padding-left: 17px;
        margin: auto;
        &:hover {
          color: grey;
          transform: scale(1.2)
        }
      }
    }

    .saved-colors {
      grid-area: colors;
      display: flex;

      div {
        background-color: pink;
        height: 50px;
        width: 50px;
        margin: 5px;
      }
    }

    .saved-palette-disp:hover {
      border: solid #3C3C3C 3px;
      cursor: pointer;
    }
  }
