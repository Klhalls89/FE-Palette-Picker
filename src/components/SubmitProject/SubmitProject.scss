.projects {
  border-top: 4px #C3C3C3 solid;
}

.saved-portion {
  display: flex;
  flex-direction: column;
  display: grid;
  grid-template-columns: repeat(2 1fr);
}

.title-input {
  font-size: 16px;
  padding-left: 9px;
}

.submit-btn {
  border: none;
  &:hover {
    cursor: pointer;
  }
} 