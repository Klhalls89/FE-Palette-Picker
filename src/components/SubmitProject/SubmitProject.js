import React, { Component } from 'react';
import Project from '../Project/Project';
import { newProject } from '../../Utility/ApiCalls';

class SubmitProject extends Component {
  constructor() {
    super();
    this.state = {
      projects: [],
      project_title: ''
    };
  }

  handleChange = (event) => {
    const { value } = event.target;
    this.setState({project_title: value});
  }


  handleSubmit = (event) => {
    event.preventDefault();
    const {project_title} = this.state;

    newProject({project_title});
    this.props.handleProject('project', project_title);
    this.setState({project_title: ''});
  }

  render() {

    return (
      <div className="projects">
        <form className="project-form" onSubmit={this.handleSubmit}>
          <input type="text" 
            className="title-input" 
            onChange={this.handleChange}
            autoComplete='off'
            value={this.state.project_title}
            placeholder='Enter Project Title'
          />
          <button>Submit</button>
        </form>
        <section className="saved-portion">
          <Project />
        </section>
      </div>
    );
  }
}

export default SubmitProject;