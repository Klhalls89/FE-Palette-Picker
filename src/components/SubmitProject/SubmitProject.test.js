import React from 'react';
import SubmitProject from './SubmitProject'
import Project from '../Project/Project';
import { shallow } from 'enzyme';

describe('SubmitProject', () => {
  let wrapper;
  let mockProjects;
  let mockHandleProject;
  let mockEvent;

  beforeEach(() => {
    let mockHandleProject = jest.fn()
     mockProjects = [{project_title: 'project one'}]
    wrapper = shallow(<SubmitProject
          handleProject={mockHandleProject}/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

    describe('handleChange', () => {
      it.skip('should set state with event change', () => {
        mockEvent = <input value="k"/>
        let event = mockEvent;
        wrapper.instance().handleChange(event)
        const state = wrapper.instance().state
        expect(state.project_title).toBe('k')
      });
    });
});