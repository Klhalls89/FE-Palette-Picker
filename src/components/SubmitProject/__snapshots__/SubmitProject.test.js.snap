// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitProjects Projects component should match the snapshot 1`] = `
<div
  className="projects"
>
  <form
    className="project-form"
    onMouseEnter={[Function]}
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="title-input"
      onChange={[Function]}
      placeholder="Enter Project Title"
      type="text"
      value=""
    />
    <button
      className="submit-btn"
    >
      Submit
    </button>
  </form>
  <section
    className="saved-portion"
  >
    <Project />
  </section>
</div>
`;
